### PR TITLE
Update Agility and Wolfs Fury

### LINKED ISSUE #334: Agility does not reduce fall damage

### LINKED ISSUE #335: agility does not cancel on attacking another player

### LINKED ISSUE #336: Agility does not give feedback when it ends.

### PR DESCRIPTION
**Pull request by StudentAlleg**
Fixes #334
Fixes #335
Fixes #336

Changes wolf fury to use getDuration for duration.

Actually implements agility, which now reduces all damage, but is cancelled on any interaction by user. Agility now correctly ends after its duration.

### CODE REVIEW

File: champions/src/main/java/me/mykindos/betterpvp/champions/champions/skills/skills/ranger/axe/Agility.java [hunk 1]
```diff
@@ -100,51 +98,60 @@ public double getCooldown(int level) {
...
     }
 
-    @UpdateEvent(delay = 500)
+    @UpdateEvent(delay = 250)
     public void onUpdate() {
-        active.removeIf(uuid -> Bukkit.getPlayer(uuid) == null);
+        active.forEach((uuid, aLong) -> {
+            Player player = Bukkit.getPlayer(uuid);
+            if (player == null) {
+                active.remove(uuid);
```
Comment (braulio-dev) [inline champions/src/main/java/me/mykindos/betterpvp/champions/champions/skills/skills/ranger/axe/Agility.java:130]: This causes a `ConcurrentModificationException` if actually reached. To do these type of in-loop removals you must use an `Iterator` or create a clone of the collection

File: champions/src/main/java/me/mykindos/betterpvp/champions/champions/skills/skills/ranger/axe/Agility.java [hunk 2]
```diff
@@ -100,51 +98,60 @@ public double getCooldown(int level) {
...
+            } else {
+                if (aLong - System.currentTimeMillis() <= 0) {
+                    deactivate(player);
```
Comment (braulio-dev) [inline champions/src/main/java/me/mykindos/betterpvp/champions/champions/skills/skills/ranger/axe/Agility.java:133]: Also another `ConcurrentModificationException`

**Review by braulio-dev — CHANGES_REQUESTED** [review]

File: champions/src/main/java/me/mykindos/betterpvp/champions/champions/skills/skills/ranger/axe/Agility.java (hunk 1 above)
Comment (StudentAlleg) [reply champions/src/main/java/me/mykindos/betterpvp/champions/champions/skills/skills/ranger/axe/Agility.java]: changed

File: champions/src/main/java/me/mykindos/betterpvp/champions/champions/skills/skills/ranger/axe/Agility.java [hunk 3]
```diff
@@ -23,19 +23,19 @@
...
+import java.util.Map;
 import java.util.UUID;
+import java.util.WeakHashMap;
 
 @Singleton
 @BPvPListener
 public class Agility extends Skill implements InteractSkill, CooldownSkill, Listener {
 
-    private final Set<UUID> active = new HashSet<>();
+    private final WeakHashMap<UUID, Long> active = new WeakHashMap<>();
```
Comment (Mykindos) [inline champions/src/main/java/me/mykindos/betterpvp/champions/champions/skills/skills/ranger/axe/Agility.java:38]: No need to use a WeakHashMap if storing a UUID, just do a regular HashMap

File: champions/src/main/java/me/mykindos/betterpvp/champions/champions/skills/skills/ranger/axe/Agility.java [hunk 4]
```diff
@@ -100,51 +100,65 @@ public double getCooldown(int level) {
...
+       while (iterator.hasNext()) {
+           Map.Entry<UUID, Long> entry = iterator.next();
+           Player player = Bukkit.getPlayer(entry.getKey());
+           if (player == null) {
+              iterator.remove();
+              continue;
+           } else {
+               if (entry.getValue() - System.currentTimeMillis() <= 0) {
+                   iterator.remove();
+                   deactivate(player);
```
Comment (Mykindos) [inline champions/src/main/java/me/mykindos/betterpvp/champions/champions/skills/skills/ranger/axe/Agility.java:141]: incredibly minor, but could you move iterator.remove() to below deactivate(player);

**Review by Mykindos — CHANGES_REQUESTED** [review]

File: champions/src/main/java/me/mykindos/betterpvp/champions/champions/skills/skills/ranger/axe/Agility.java (hunk 3 above)
Comment (StudentAlleg) [reply champions/src/main/java/me/mykindos/betterpvp/champions/champions/skills/skills/ranger/axe/Agility.java]: Changed

**Review by Mykindos — APPROVED** [review]

**Pull request merged**